web-api/src/lambdas/streams/processStreamRecordsLambda.ts

```diff
@@ -1,86 +1,42 @@
-import {
-  batchGet,
-  getTableName,
-} from '../../persistence/dynamodbClientService';
 import { createApplicationContext } from '../../applicationContext';
 import type { DynamoDBRecord, DynamoDBStreamEvent } from 'aws-lambda';
 
-/**
- *
- */
-async function putEventHistory(
-  applicationContext: IApplicationContext,
-  eventID: string,
-) {
-  const twentyFourHours = 24 * 60 * 60;
-
-  await applicationContext
-    .getDocumentClient({
-      useMasterRegion: true,
-    })
-    .put({
-      Item: {
-        pk: `stream-event-id|${eventID}`,
-        sk: `stream-event-id|${eventID}`,
-        ttl: Math.floor(Date.now() / 1000) + twentyFourHours,
-      },
-      TableName: getTableName({
-        applicationContext,
-      }),
-    })
-    .promise();
-}
-
-/**
- *
- */
-async function getEventHistories(
-  applicationContext: IApplicationContext,
-  records: DynamoDBRecord[],
-) {
-  return (await batchGet({
-    applicationContext,
-    keys: records
-      .map(record => record.eventID)
-      .map(eventID => ({
-        pk: `stream-event-id|${eventID}`,
-        sk: `stream-event-id|${eventID}`,
-      })),
-  })) as { pk: string; sk: string }[];
-}
+const applicationContext = createApplicationContext({});
+const deploymentTimestamp: number = Number(process.env.DEPLOYMENT_TIMESTAMP!); // epoch seconds
+
+const shouldProcessRecord = (record: DynamoDBRecord): boolean => {
+  if (
+    record &&
+    'dynamodb' in record &&
+    record.dynamodb &&
+    'ApproximateCreationDateTime' in record.dynamodb &&
+    typeof record.dynamodb.ApproximateCreationDateTime !== 'undefined'
+  ) {
+    // StreamRecord objects from local dynamodb have an ApproximateCreationDateTime with a type of Date
+    // StreamRecord objects from AWS dynamodb have an ApproximateCreationDateTime with a type of number (epoch seconds)
+    // here we'll handle both types and convert to epoch seconds if necessary
+    const approximateCreationDateTime: number =
+      Number(record.dynamodb.ApproximateCreationDateTime) > 9999999999
```

Comment:
This is a pretty hacky way to determine if our number is epoch seconds or epoch millis. I am open to other suggestions.

Reply:
what about checking the types of what you're dealing with?

```
if (record?.dynamodb?.ApproximateCreationDateTime) {
  const {
    ApproximateCreationDateTime,
  }: { ApproximateCreationDateTime?: any } = record.dynamodb;

  let approximateCreationDateTime: number;

  if (ApproximateCreationDateTime instanceof Date) {
    approximateCreationDateTime = Math.floor(
      ApproximateCreationDateTime.getTime() / 1000,
    );
  } else if (typeof ApproximateCreationDateTime === 'number') {
    approximateCreationDateTime = ApproximateCreationDateTime;
  } else {
    throw new Error(
      `Could not handle ApproximateCreationDateTime: ${typeof ApproximateCreationDateTime}`,
    );
  }
  
  // ...

}
```

Reply:
Love it, updated